presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -46,15 +51,33 @@
         private final PlanNodeId planNodeId;
         private final Step step;
         private final List<AccumulatorFactory> accumulatorFactories;
+        private final List<Type> sourceTypes;
+        private final List<List<Integer>> orderByChannels;
+        private final List<List<SortOrder>> orderings;
         private final List<Type> types;
+        private final Optional<PagesIndex.Factory> pagesIndexFactory;
         private boolean closed;
 
-        public AggregationOperatorFactory(int operatorId, PlanNodeId planNodeId, Step step, List<AccumulatorFactory> accumulatorFactories)
+        public AggregationOperatorFactory(
+                int operatorId,
+                PlanNodeId planNodeId,
+                Step step,
+                List<AccumulatorFactory> accumulatorFactories,
+                List<Type> sourceTypes,
+                List<List<Integer>> orderByChannels,
+                List<List<SortOrder>> orderings,
+                Optional<PagesIndex.Factory> pagesIndexFactory)
         {
             this.operatorId = operatorId;
             this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
             this.step = step;
             this.accumulatorFactories = ImmutableList.copyOf(accumulatorFactories);
+            this.sourceTypes = requireNonNull(sourceTypes);
+            requireNonNull(orderByChannels, "orderByChannels is null");
+            this.orderByChannels = ImmutableList.copyOf(orderByChannels.stream().map(ImmutableList::copyOf).collect(toImmutableList()));
+            requireNonNull(orderings, "orderings is null");
+            this.orderings = ImmutableList.copyOf(orderings.stream().map(ImmutableList::copyOf).collect(toImmutableList()));
```

Comment:
* as above
* as elsewhere: does `orderings` needs to be same length as `orderByChannels` ? Do their elements (the lists) need to be pairwise same length?